src/symbol/placement.js

```diff
@@ -647,6 +709,15 @@ export class Placement {
             }
         }
 
+        if (showCollisionBoxes && bucket.bucketInstanceId in this.collisionCircleArrays) {
+            const circleArray = this.collisionCircleArrays[bucket.bucketInstanceId];
+
+            // Store viewport and inverse projection matrices per bucket
+            mat4.invert(circleArray.invProjMatrix, posMatrix);
+            circleArray.viewportMatrix = this.collisionIndex.getViewportMatrix();
+            circleArray.bucket = bucket;
```

Comment:
As you asked, holding onto this reference here might not be great since it's possible for the bucket to no longer be used when `commit()` is called. While I don't think it should error (even if the bucket is not used the reference should be valid enough) it might be cleaner to do it elsewhere.

Would it be possible to make the changes to the buffer in either within this method, or later within `updateLayerOpacities(...)` when we update the bucket's buffers that are actually used for rendering?

Reply:
I've removed the bucket reference and moved the update logic to `updateLayerOpacities()`

Reply:
Looks good!